runtime/src/lib.rs

```diff
@@ -1871,6 +1914,27 @@ mod tests {
         test_panic_aborts_spawn(executor);
     }
 
+    #[test]
+    fn test_deterministic_panic_aborts_spawn_caught() {
+        let cfg = deterministic::Config::default().with_catch_panics(true);
+        let executor = deterministic::Runner::new(cfg);
+        test_panic_aborts_spawn(executor);
+    }
+
+    #[test]
+    #[should_panic(expected = "boom")]
```

Comment:
panics on the first boom but you're not sure if its 1 2 or 3?

Reply:
It doesn't matter which "boom" it is. What we care is that it isn't an internal panic (that's what the regression test is for basically).